docs/misc/optional-dependencies.txt

```diff
@@ -0,0 +1,73 @@
+=====================
+Optional dependencies
+=====================
+
+The following optional dependencies are required to provide support for some of
+the functionality within Django. For convenience it is possible to install
+these dependencies along with django, e.g. ``pip install django[gis,mysql]``.
+
++------------+-------------------+---------------------------------+
+| Category   | Extra Name        | Requirements                    |
```

Comment:
I think we have to add a description to explain what's the function of an optional dependency. For some of them the motivation is clear (e.g. `postgresql` is required to connect to a PostgreSQL database) but for other dependencies the motivation is not obvious (e.g. why use `numpy` in GIS category ?).

Reply:
Thanks. I think I had that on my list of things to consider. I'm also planning to have a bit of a rethink into how much we do here - certainly document all of the possible dependencies, but probably have fewer extras defined.